src/controllers/petsController.js

```diff
@@ -168,11 +170,27 @@ const getPet = async (req, res, next) => {
 
 const deletePet = async (req, res, next) => {
   log.debug("Calling deletePet...");
+
+  //get image path
+  await getPetById(req.app.get('db'), req.params.petID)
+    .then((dbResponse) => {
+      if (dbResponse.length == 0) {
+        return res.status(404).json({ error: "Pet not found." });
+      }
+      removeImageFile(dbResponse[0].Picture)
```

Comment:
If we want to be extra careful, I would suggest moving the `deletePetById` part before `removeImageFile(dbResponse[0].Picture)`, to make sure we only delete the image after we've successfully deleted a pet.

Reply:
thank you! fixed!